i18n-embed-fl/src/lib.rs

```diff
@@ -165,10 +172,57 @@ struct DomainSpecificData {
     _assets: FileSystemAssets,
 }
 
-fn domains() -> &'static dashmap::DashMap<String, DomainSpecificData> {
-    static DOMAINS: OnceLock<dashmap::DashMap<String, DomainSpecificData>> = OnceLock::new();
+#[derive(Default)]
+struct DomainsMap {
+    #[cfg(not(feature = "performance"))]
+    map: RwLock<HashMap<String, Arc<DomainSpecificData>>>,
+
+    #[cfg(feature = "performance")]
+    map: dashmap::DashMap<String, DomainSpecificData>,
+}
+
+impl DomainsMap {
+    #[cfg(not(feature = "performance"))]
```

Comment:
You can use the `#[cfg(not(feature = "performance"))]` annotation also in the function body. I think it makes sense to share the function signature and just apply this logic to the body. e.g.
```rust
fn get(&self, domain: &String) -> Option<Arc<DomainSpecificData>> {
    #[cfg(not(feature = "performance"))]
    match self.map.read().unwrap().get(domain) {
        None => None,
        Some(data) => Some(data.clone()),
    }
    #[cfg(feature = "performance")]
    self.map.get(domain)
}
```

Though even better is the aproach I outlined above where you just create a `RwLockHashMap` that just behaves like it was a `DashMap` or `HashMap+Sync`.

Reply:
I'd love to share the function signature, but after trying I realize that the `Ref<String, DomainSpecificData>` returned by `dashmap` is also holding a `RwLockReadGuard`, so they cannot share the same signature unfortunately, unless there's a way to cheat the return type. I tried a generic method, but it seems like a no-go here, too.

I'm going to rethink this a bit.